### PR TITLE
Unbreak mockito tests for cache

### PR DESCRIPTION
**Pull request by iambmelt**
Something was broken between incrementing the mockito version and reversion back/forth between Java 1.7 & 1.8 source compat. Something inside of Mockito needs to be desugared when using `Mockito.openMocks()` so for now switching away from an annotation based test implementation

### CODE REVIEW

File: common/src/androidTest/java/com/microsoft/identity/common/BrokerOAuth2TokenCacheTest.java [hunk 1]
```diff
@@ -224,11 +223,13 @@ public void tearDown() throws Exception {
             mFociCredentialCache.clearAll();
         }
 
-        for (final IAccountCredentialCache cache : mOtherAppCredentialCaches) {
-            cache.clearAll();
-        }
+        if (null != mOtherAppCredentialCaches)
```
Comment (iambmelt) [inline common/src/androidTest/java/com/microsoft/identity/common/BrokerOAuth2TokenCacheTest.java:226]: Remove this

File: common/src/androidTest/java/com/microsoft/identity/common/BrokerOAuth2TokenCacheTest.java [hunk 2]
```diff
@@ -224,11 +223,13 @@ public void tearDown() throws Exception {
...
+            for (final IAccountCredentialCache cache : mOtherAppCredentialCaches) {
+                cache.clearAll();
+            }
 
-        mApplicationMetadataCache.clear();
+        if (null != mApplicationMetadataCache)
```
Comment (iambmelt) [inline common/src/androidTest/java/com/microsoft/identity/common/BrokerOAuth2TokenCacheTest.java:231]: Remove this

File: common/src/androidTest/java/com/microsoft/identity/common/BrokerOAuth2TokenCacheTest.java [hunk 3]
```diff
@@ -125,7 +119,12 @@
     public void setUp() throws Exception {
         super.setUp();
 
-        MockitoAnnotations.initMocks(this);
+        //MockitoAnnotations.openMocks(this);
```
Comment (iambmelt) [inline common/src/androidTest/java/com/microsoft/identity/common/BrokerOAuth2TokenCacheTest.java:122]: Remove comment

Comment (iambmelt) [inline common/src/androidTest/java/com/microsoft/identity/common/BrokerOAuth2TokenCacheTest.java:122]:
```suggestion
```

**Review by AdamBJohnsonx — APPROVED** [review]
If that's what it takes, fine.  Can we indicate to ourselves that we can revisit this once we get to Java8 support?

**Comment by iambmelt** [comment]
> If that's what it takes, fine. Can we indicate to ourselves that we can revisit this once we get to Java8 support?

Created a backlog item for it:
https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1291599

Thanks for reviewing 👍

**Pull request merged**